### PR TITLE
feat: lingua pretraining dataset

### PR DESCRIPTION
**Pull request by adil-a**
This PR adds the Lingua (jsonl format) dataset consumption feature in AutoModel

### CODE REVIEW

File: nemo_automodel/components/datasets/llm/jsonl_dataset.py [hunk 1]
```diff
@@ -0,0 +1,220 @@
...
+                    break
+            elif not batch_available:
+                # In non-distributed or infinite mode, just stop when data is exhausted
+                break
+
+            # Process and yield the batch if available
+            if batch_available:
+                batch = torch.from_numpy(batch)
+                bs, s, _ = batch.shape
+                loss_mask = torch.ones((bs, s), dtype=torch.int64)
```
Comment (akoumpa) [inline nemo_automodel/components/datasets/llm/jsonl_dataset.py:175]: can we drop the loss_mask and modify the labels instead?

File: nemo_automodel/components/datasets/llm/jsonl_dataset.py [hunk 2]
```diff
@@ -0,0 +1,220 @@
...
+            n_views,
+            split,
+            not infinite,
+        )
+
+        # Create the context stack to manage the dataloader lifecycle and build the dataloader
+        self._build_dataloader()
+
+        # Check if distributed training is initialized
+        self._is_distributed = torch.distributed.is_available() and torch.distributed.is_initialized()
```
Comment (akoumpa) [inline nemo_automodel/components/datasets/llm/jsonl_dataset.py:111]: can the user pass `dp_group=None` and have `torch.distributed` initialized?
